### DISCUSSION TITLE
Visual Studio Mac M1: Crash entering input field on iOS simulator iPad (10th gen) and iPad Pro (6th gen) compiled for iOS 17.4

### DISCUSSION THREAD

**Discussion by Yveske1966** [AvaloniaUI/Avalonia#15309] [Q&A] [answered]
Visual Studio Mac 17.6.11 (build 400)
Avalonia 11.1.0-beta1

Compiling for iOS 17.0 and iOS 17.2 runs fine and shows onscreen keyboard, on iOS 17.4 the app crashes immediately on entering the input field.

**Comment by timunie** [comment]
try to run MobileSandbox under src/Samples - Probably you can find the root cause and file a fix for it?

**Comment by DrWenz** [comment]
@Yveske1966 did you found the issue?

**Reply by Yveske1966** [reply]
No, I haven't. I was hoping for a new version of Avalonia to solve the problem.

**Reply by Yveske1966** [reply]
May I ask if you used a physical phone/tablet when you encountered the error? I used a simulator on Mac.
May I ask if you used a Windows or Mac computer to connect to the physical tablet (if any)?
It may help the developers to pinpoint the problem.

**Comment by maxkatz6** [comment]
Any logs? Stack traces?

**Reply by Yveske1966** [reply]
Let me know if I can be of further assistance

Crash report in attachment
[Crash report IOS 17.4.txt](https://github.com/user-attachments/files/15585278/Crash.report.IOS.17.4.txt)

**Reply by workgroupengineering** [reply]
hit could be the same error of https://github.com/dotnet/maui/issues/22323

**Reply by Yveske1966** [reply]
I can use pickers (date and string) in my program using XCode 15.4

**Reply by workgroupengineering** [reply]
From the stack view it appears that there is a problem with the localization of the decimal separator. If you have string formats, try elmiating them to see if there is a problem.

```bash
11  libicucore.A.dylib            	       0x122920d6f icu::Locale::setToBogus() + 43
12  libicucore.A.dylib            	       0x12292284a icu::Locale::operator=(icu::Locale const&) + 30
13  libicucore.A.dylib            	       0x122a8e590 icu::number::UnlocalizedNumberFormatter::locale(icu::Locale const&) && + 14
14  libicucore.A.dylib            	       0x122a3ceb7 icu::DecimalFormat::touch(UErrorCode&) + 257
15  libicucore.A.dylib            	       0x122a3d52b icu::DecimalFormat::DecimalFormat(icu::UnicodeString const&, icu::DecimalFormatSymbols*, UNumberFormatStyle, UErrorCode&) + 297
16  libicucore.A.dylib            	       0x122aaa93f icu::NumberFormat::makeInstance(icu::Locale const&, UNumberFormatStyle, signed char, UErrorCode&) + 1339
17  libicucore.A.dylib            	       0x122aaa2ad icu::LocaleCacheKey<icu::SharedNumberFormat>::createObject(void const*, UErrorCode&) const + 81
18  libicucore.A.dylib            	       0x1229d08dc icu::UnifiedCache::_get(icu::CacheKeyBase const&, icu::SharedObject const*&, void const*, UErrorCode&) const + 106
19  libicucore.A.dylib            	       0x122aaae78 0x12291f000 + 1621624
20  libicucore.A.dylib            	       0x122aaa3ad 0x12291f000 + 1618861
21  libicucore.A.dylib            	       0x122aa9f56 icu::NumberFormat::createInstance(icu::Locale const&, UNumberFormatStyle, UErrorCode&) + 42
22  libicucore.A.dylib            	       0x122921e8f unum_open + 159
```

**Reply by jp2masa** [reply, marked as answer]
- https://github.com/xamarin/xamarin-macios/issues/20421#issuecomment-2047839085
- https://github.com/dotnet/runtime/issues/98941
